### DISCUSSION TITLE
Wagmi's equivalent to Ethers await tx.wait()

### DISCUSSION THREAD

**Discussion by fseitun-ac** [wevm/wagmi#534] [Q&A]
On ethers I can wait for a tx to be mined by calling `await tx.wait()` is there an equivalent in wagmi? I'm using  ```useWaitForTransaction({ hash: data?.hash, wait: data?.wait })``` to no avail.
    
Can anyone point me in the right direction?

Just in case there's a more declarative approach: I want to show the user which token he minted, and to do that i have to await for the tx to be mined.

Thanks!

**Comment by tmm** [comment]
`useWaitForTransaction` should be what you want. Seems like some others are confused/running into issues on how to use it.

Can you put together a minimal reproducible example of it not working and open an issue? Happy to look into it.

**Reply by fseitun-ac** [reply]
I'll answer myself: `useContractWrite` has an async function that can be "awaited"
```
const { writeAsync } = useContractWrite(/*...*/)
/*...*/
async function mintToken() {
  const tx = await writeAsync();
  await tx.wait();
  /* now tx has been mined */
}
```

**Reply by fseitun-ac** [reply]
> `useWaitForTransaction` should be what you want. Seems like some others are confused/running into issues on how to use it.
> 
> Can you put together a minimal reproducible example of it not working and open an issue? Happy to look into it.

I think the actual problem isn't with `useWaitForTransaction`. It's with `writeAsync` not being documented (yet). If you want i'll be glad to help!
